### ISSUE TITLE
svelte/no-template-shadow

### ISSUE THREAD

**Issue by Shyam-Chen** [sveltejs/eslint-plugin-svelte#831]
### Motivation

https://eslint.vuejs.org/rules/no-template-shadow.html

### Description

Disallow variable declarations from shadowing variables declared in the outer scope

### Examples

```svelte
<script lang="ts">
  let foo = $state(['bar', 'baz']);
</script>

<!-- ✓ GOOD -->
<ul>
  {#each [1, 2, 3] as num (num)}
    <li>{num}</li>
    <ul>
      {#each [11, 22, 33] as subNum (subNum)}
        <li>{subNum}</li>
      {/each}
    </ul>
  {/each}
</ul>


<!-- ✗ BAD -->
<ul>
  {#each [1, 2, 3] as num (num)}
    <li>{num}</li>
    <ul>
      {#each [11, 22, 33] as num (num)}
        <li>{num}</li>
      {/each}
    </ul>
  {/each}
</ul>

<ul>
  {#each [1, 2, 3] as foo (foo)}
    <li>{foo}</li>
  {/each}
</ul>
```

**Comment by ota-meshi** [comment]
This plugin works differently from eslint-plugin-vue, so you can check it with the `no-shadow` rule.

https://eslint.org/docs/latest/rules/no-shadow


[Online Demo](https://eslint-online-playground.netlify.app/#eNq9VNuO0zAQ/ZXBIG1b5VJAvHS7qwUWeILy3hTJm7jdlMSObGfpUvUreOCFr+NLGMfOrS0FgaAPTTxzzpmLZ7IlSsbh2/uXIi8EZ1wH6o5lmpEJmapYpoWGjPLVRUS0ishlxAHCEVNZyjVw4atbmohPE4gIk1LIiIxCA8mYhqUQcAGPlKaaDeZnN1SeeYCPz2eL4XnEp6GVR0k8PPB9+P7tC7yZza7B99E4LbMq2vYho/EtzB978MSDpwugCniZwwD/hjuDAJhm6eUWz7tpiG/O5vjm12gYEaNiZVR5884o2Wct1ghac1cTlUKjVIcN6xxrq7N0SvoKL57/v4r2ZX4m9fuFnEra3PAA/3pJ47mN1NcjHrGTE8SCL9NVsFY4ZikOntSwRj0pcpykKwsK1zhwOCjOb8fyfVauUl4jLdAvKqNvEUc4VComa461+TW18nU5Wrnpdnh9XzA7qY5TgdmmAidsScusJbnCBqb2tXInFUgWizxnPGGJZ1xBEOwxjmK6Jde4eUSWGdVhBx6RRcXY2ltYphlTE0DgaBSOnEhEPKylPVkGLirudklXbFboVHCkOREA25lJr4eOVDsPSS2tYwJsj/lCXB20Mqzb3wgD7DrvbKMlfY31vNpoxpUNhoUdlNHSqlG0h1P9wP73W1IZ/nlXmlu3hj/O3qxGL3tj+Kvsm8jDc1zTgsYfUQFlBccVrQTsKkTEXGUukjIzQ2UdCbu7ZoUZRh6nDFNpYva32VBxeJnCJXJx6yU+6dzf8KPQrvPDs2AcjP3xgXt/748qtXP6S0DzTejiqlZGfEd2PwBtwl1L)
